### PR TITLE
Cast proper types when loading resource data

### LINKED ISSUE #82: Cast proper types when loading resource data?
I was following the [Python quick start guide](http://frictionlessdata.io/guides/quick-start-in-python/) and playing with the [periodic table](http://frictionlessdata.io/example-datasets/periodic-table/datapackage.json) DP. 

I checked the resource metadata and the schema correctly describes `atomic number` and `atomic mass` as `integer` and `number` respectively, but when I tried to do:

```
> print [e['name'] for e in dp.resources[0].data if e['atomic number'] < 10]
[]
```

I noticed that the data has these fields as strings:

```
> print dp.resources[0].data[0]
{u'metal or nonmetal?': u'nonmetal', u'symbol': u'H', u'atomic number': u'1', u'name': u'Hydrogen', u'atomic mass': u'1.00794'}
```

Not sure about how the parsing works but would it make sense to use to information on the schema to cast the values appropriately?

### PR DESCRIPTION
**Pull request by akariv**
Fixes #82

### CODE REVIEW

**Comment by vitorbaptista** [comment]
@akariv Is this good to go?

**Comment by akariv** [comment]
@vitorbaptista as far as I can tell...

File: datapackage/resource.py [hunk 1]
```diff
@@ -319,15 +322,28 @@ def _is_url(path):
...
+        if schema is not None:
+            self._schema = jsontableschema.model.SchemaModel(schema)
 
     def __iter__(self):
         return self
 
     def __next__(self):
         row = next(self._tabulator_iter)
-        return dict(zip(row.headers, row.values))
+        row = dict(zip(row.headers,row.values))
```
Comment (vitorbaptista) [inline datapackage/resource.py]: Please add a whitespace between `row.headers` and `row.values`

File: datapackage/resource.py [hunk 2]
```diff
@@ -5,9 +5,12 @@
 
 
 import os
+import json
 import six
 import six.moves.urllib as urllib
+from future.utils import raise_with_traceback
```
Comment (vitorbaptista) [inline datapackage/resource.py]: I've never used `raise_with_traceback`, but it sounds interesting. Over `datapackage-py`, we've been using `six.raise_from` as a Python-version independent way of wrapping exceptions (like Py3's `raise XXX from exp`), but I wonder if `raise_with_traceback` is better (of it it's the same, in which case we should use only one).

**Comment by vitorbaptista** [comment]
@akariv I'm sorry for taking so long to review this. Could you check my comments, rebase with master and push a new version? I'm hoping to merge it this week.

File: datapackage/resource.py (hunk 1 above)
Comment (akariv) [inline datapackage/resource.py]: done

File: datapackage/resource.py (hunk 2 above)
Comment (akariv) [inline datapackage/resource.py]: done

**Comment by vitorbaptista** [comment]
Thanks, @akariv!

**Pull request merged**